File: codingbat/java/last2.java
public int last2(String str) {
  if (str.length() <= 2) return 0;
  
  String target = str.substring(str.length() - 2);
  int targetCount = 0;
  for (int i = 0; i < str.length() - 2; i++) {
    if (str.substring(i, i + 2).equals(target)) targetCount++;
  }
  return targetCount;
}

File: codingjs/last2.js
function last2(str: string): number {
  if (str.length <= 2) return 0;
  const word = str.substring(str.length - 2);
  return last2Helper(str, word, 0);
}
function last2Helper(target: string, word: string, count: number): number {
  if (target.length <= 0) return count - 1;
  if (target.substr(0, 2) === word) count++;
  return last2Helper(target.substring(1), word, count);
}